### PR TITLE
modules: hal: nordic: Integrate nrfx 2.10

### PR DESCRIPTION
**Pull request by awojasinski**

### CODE REVIEW

**Review by anangl — DISMISSED** [review]

File: modules/hal_nordic/nrfx/nrfx_config.h [hunk 1]
```diff
@@ -9,6 +9,12 @@
 
 #include <zephyr/devicetree.h>
 
+/*
+ * NRFX API version 2.9 flag.
+ * When the flag is set NRFX API is compatible with the previous NRFX release.
+ */
+#define NRFX_CONFIG_API_VER_2_9 1
```
Comment (nika-nordic) [inline modules/hal_nordic/nrfx/nrfx_config.h:16]: shouldn't we switch to nrfx 2.10 API immediately?

Comment (awojasinski-nordicsemi) [reply modules/hal_nordic/nrfx/nrfx_config.h:16]: No, zephyr and its modules are not aligned with API changes. When proper changes will be applied then `NRFX_CONFIG_API_VER_2_10` symbol should be set to 1

**Review by microbuilder — DISMISSED** [review]
You'll need to squash some of these merge commits before this can be merged in.

**Comment by awojasinski-nordicsemi** [comment]
@anangl @microbuilder PR to `hal_nordic` merged and revision updated. Ready to be merged

**Review by anangl — APPROVED** [review]

**Review by hubertmis — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Review by nika-nordic — APPROVED** [review]

**Pull request merged**